### DISCUSSION TITLE
How To Add Multiple Where Clause ?

### DISCUSSION THREAD

**Discussion by rahat14** [simolus3/drift#1880] [Q&A]
i need to add multiple where query using or/AND. How to achieve this in drift

**Comment by simolus3** [comment]
If you need an `AND` in SQL, you can just call `where` multiple times and drift will and them together for you. E.g. `select(users)..where((row) => row.id.equalsValue(3))..where((row) => row.name.length.isGreaterThanValue(12))` will give you `WHERE id = 3 AND LENGTH(name) > 12`.

If you want to use an `OR` (or an `AND`) directly, you can use operators defined on `Expression`:

```dart
where((row) => row.id.equalsValue(3) | row.name.length.isGreaterThanValue(12)) // id = 3 OR LENGTH(name) > 12
where((row) => row.id.equalsValue(3) & row.name.length.isGreaterThanValue(12)) // id = 3 AND LENGTH(name) > 12
```
